### ISSUE TITLE
[Allure-Playwright] More tests are reported than were actually run

### ISSUE THREAD

**Issue by qakos** [allure-framework/allure-js#560]
**Describe the bug**
More tests are reported than were actually run. For example, 79 tests were run, of which 73 were successful, 3 failed and only 3
after restarting. The Allure report included 85 tests, of which 73 were successful, 3 broken and 3 failed.

**Expected behavior**
The report should have the same number of tests as when the tests were run (79)

**Screenshots**

![22-12-22_01](https://user-images.githubusercontent.com/93794669/209233115-3df8fcb3-07ca-4776-8461-a8a569600db7.jpg)
![22-12-22_02](https://user-images.githubusercontent.com/93794669/209233118-b8e449b4-428b-4301-badc-f7270e294542.jpg)

**Desktop (please complete the following information):**
 - Playwright Version: 1.29.0, 1.29.1
 - Operating System: Windows, Linux, Mac
 - Node.js version: 18.12.1
 - Browser: Chromium, Firefox, WebKit
 - Allure: 2.0.0-beta.21

**Comment by vovsemenv** [comment]
Hello, can you please send you playwright config? I imagine there is something with retries

**Comment by vovsemenv** [comment]
@qakos ping

**Comment by qakos** [comment]
// @ts-check
const { devices } = ***@***.***/test');

/**
* Read environment variables from file.
* https://github.com/motdotla/dotenv
*/
require('dotenv').config();

/**
* @see https://playwright.dev/docs/test-configuration
* @type ***@***.***/test').PlaywrightTestConfig}
*/
const config = {
fullyParallel: true,
globalSetup: require.resolve('./global-setup'),
testDir: './tests/',
testMatch: ['/tests/**/*.test.js'],
// testMatch: ['/tests/**/*.test.js'],
/* Maximum time one test can run for. */
timeout: 60 * 1000,
expect: {
/**
* Maximum time expect() should wait for the condition to be met.
* For example in `await expect(locator).toHaveText();`
*/
timeout: 6000,
},
/* Fail the build on CI if you accidentally left test.only in the source
code. */
forbidOnly: !!process.env.CI,
/* Retry on CI only */
retries: process.env.CI ? 1 : 1,
// repeatEach: 6,
/* Opt out of parallel tests on CI. */
workers: process.env.CI ? 6 : 2,
/* Reporter to use. See https://playwright.dev/docs/test-reporters */
reporter: [
['list'],
['allure-playwright'],
['html', { open: 'never', outputFolder: 'my-report' }],
],
/* Shared settings for all the projects below. See
https://playwright.dev/docs/api/class-testoptions. */
use: {
/* Maximum time each action such as `click()` can take. Defaults to 0 (no
limit). */
actionTimeout: 0,
/* Base URL to use in actions like `await page.goto('/')`. */
// baseURL: 'http://localhost:3000',

/* Collect trace when retrying the failed test. See
https://playwright.dev/docs/trace-viewer */
trace: process.env.CI ? 'on-first-retry' : 'retain-on-failure',
screenshot: 'only-on-failure',
// headless: false,
video: process.env.CI ? 'on-first-retry' : 'off',
},

/* Configure projects for major browsers */
projects: [
{
name: 'chromium',
use: {
...devices['Desktop Chrome'],
viewport: { width: 1280, height: 768 },
},
},

{
name: 'firefox',
use: {
...devices['Desktop Firefox'],
viewport: { width: 1280, height: 768 },
},
},

{
name: 'webkit',
use: {
...devices['Desktop Safari'],
viewport: { width: 1280, height: 768 },
},
},

/* Test against mobile viewports. */
// {
// name: 'Mobile Chrome',
// use: {
// ...devices['Pixel 5'],
// },
// },
// {
// name: 'Mobile Safari',
// use: {
// ...devices['iPhone 12'],
// },
// },

/* Test against branded browsers. */
// {
// name: 'Microsoft Edge',
// use: {
// channel: 'msedge',
// },
// },
// {
// name: 'Google Chrome',
// use: {
// channel: 'chrome',
// },
// },
],

/* Folder for test artifacts such as screenshots, videos, traces, etc. */
// outputDir: 'test-results/',

/* Run your local dev server before starting the tests */
// webServer: {
// command: 'npm run start',
// port: 3000,
// },
};

module.exports = config;


пт, 23 груд. 2022 р. о 16:27 Vladimir Semenov ***@***.***>
пише:

**Comment by qakos** [comment]
> Hello, can you please send you playwright config? I imagine there is something with retries

Here is my PW config, retries were working well as need, but in one moment something goes wrong..

**Comment by qakos** [comment]
@vovsemenv ping

**Comment by Yurati** [comment]
hey, I'm experiencing the same bug. After upgrade to Playwright v1.29.0 retries are marked as separate tests in my case as well.

**Comment by vovsemenv** [comment]
Guys, thank you for participation. Found out there is a regression in newer version of playwright adapter. Reproduced only in `allure-report` if you using allure TestOps everything should be fine

**Comment by vovsemenv** [comment]
PR ready: https://github.com/allure-framework/allure-js/pull/563

**Comment by qakos** [comment]
@vovsemenv thank you!

**Comment by qakos** [comment]
@vovsemenv after installing update, I have the same problem: 
<img width="406" alt="28-12-22__" src="https://user-images.githubusercontent.com/93794669/209807917-25362acc-2dfe-4ecd-ab10-6e7bfa49611d.png">
this test is flaky, but in report it's marked as failed. 
In my test suit I have 36 tests, in result of test run I have: 1 failed, 8 flaky, 27 passed.
In report, I have 45 tests with 35 passed and 10 failed
<img width="406" alt="28-12-22__" src="https://user-images.githubusercontent.com/93794669/209808590-95cf72f3-9240-44ee-b97a-1307661757bd.png">
<img width="401" alt="28-12-22__" src="https://user-images.githubusercontent.com/93794669/209808627-002d9a97-d80b-499a-a5d6-9a6dd6c01142.png">
All project settings the same. I just update allure-playwright to 2.0.0-beta.22

**Comment by vovsemenv** [comment]
I published release just now) please use `2.0.0-beta.23` issue will be fixed @qakos

**Comment by qakos** [comment]
@vovsemenv thank you!

**Comment by qakos** [comment]
@vovsemenv Nothing has changed
<img width="307" alt="28-12-22__" src="https://user-images.githubusercontent.com/93794669/209812159-8b7f876d-d2c5-43e9-91ef-990c0b3edaf2.png">
Actual result: 2 flaky, 34 passed. In report 38 tests: 2 failed, 36 passed.
Expected results in report: 36 passed.

**Comment by vovsemenv** [comment]
🤔 Are you sure that you actually using `2.0.0-beta.23` ? please try reinstalling.
I have code like this with 1 retry and everything looks fine

```ts
import { test, expect } from "@playwright/test";
test("all-completed", async ({ page }) => {
  const n = Math.random();
  expect(n > 0.5).toBe(false);
});
```

<img width="1529" alt="image" src="https://user-images.githubusercontent.com/20096510/209813589-bcdfce6d-603b-4b89-84d7-56086e936822.png">

**Comment by vovsemenv** [comment]
please show output of `npm list allure-playwright @playwright/test`

**Comment by qakos** [comment]
I delete node_modules and install all.
Nothing has changed

**Comment by vovsemenv** [comment]
Thanks, looks like some other bug. Will investigate.

**Comment by Yurati** [comment]
For me it was fixed, thanks! I'm using also Playwright 1.29.1 with the newest plugin & generate report using `allure-commandline` 2.20.1

**Comment by qakos** [comment]
@vovsemenv In GitLab report working well, so I think it's something with my computer. Thank you for quick fix!
